### PR TITLE
Revendor k8s.io/utils and use diff+fields from there

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/kubernetes/utils/pull/82

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go [hunk 1]
```diff
@@ -16,76 +16,12 @@ limitations under the License.
...
 // Path represents the path from some root to a particular field.
-type Path struct {
-	name   string // the name of this field or "" if this is an index
-	index  string // if name == "", this is a subscript (index or map key) of the previous element
-	parent *Path  // nil if this is the root element
-}
+type Path = field.Path
 
 // NewPath creates a root Path object.
 func NewPath(name string, moreNames ...string) *Path {
```
Comment (thockin) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go]: Why leave this shim in place?  I know there are a lot of callers of this, but it seems like something gorename (or sed) could fix?

**Review by thockin — COMMENTED** [review]
I feel like I've been doing a ot of these reviews lately.  I'd like to get a few more deps reviewers in the loop.  Any objections?  @cblecker @sttts and I are the only explicit ones listed.

File: staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go]: This affects bascially all 3rdparty API developers. You see it in our code with a hundred uses or so.

**Pull request closed without merging**